bokulich-lab/q2-moshpit
# ENH: add action to generate human pangenome index and filter against it

This PR adds an action allowing to:
1. filter reads against the provided reference (mirrors `filter-reads` action from `q2-quality-control`)
2. if no reference was provided, build one from the human pangenome sequence and Grch38 genome, filter reads against it and return both, the filtered reads and the built reference.

If anyone wants to test this before merging, you'll need to install the missing dependencies: _gfatools_, _seqtk_ and _q2-quality-control_.

To test, execute the following:
```
qiime moshpit filter-reads-pangenome \
  --i-reads <some reads artifact> \
  --p-n-threads N \
  --o-filtered-reads filtered.qza \
  --o-reference-index new-index.qza
```
After the first go, you can then try to use the index obtained through that action to test whether things work when an index is provided:
```
qiime moshpit filter-reads-pangenome \
  --i-reads <some reads artifact> \
  --i-index new-index.qza \
  --p-n-threads N \
  --o-filtered-reads filtered.qza \
  --o-reference-index newer-index.qza
```
After the second run `new-index.qza` and `newer-index.qza` should be the same artifact.
